mapper/orm/src/main/java/org/hibernate/search/mapper/orm/outbox/impl/OutboxTableIndexerExecutor.java

```diff
@@ -63,11 +63,23 @@ public void start() {
 	}
 
 	private void run() {
-		try ( Session session = mapping.sessionFactory().openSession() ) {
-			processOutboxRetrialsEntities( session );
+		if ( mapping.sessionFactory().isClosed() ) {
```

Comment:
I'm a bit surprised. Hibernate Search should shut down when `SessionFactoryObserver#sessionFactoryClosing` is closed, so *before* the session factory is actually closed.
When Hibernate Search gets closed, the automatic indexing stategy's `stop` method should be called, which should stop any event processing *before* the session factory is actually closed.

So you shouldn't have such problem... unless the `stop` method doesn't actually stop processing. And I think that's the case here, because of the loop you have in the event processor (`while (!events.isEmpty())`).

When you're done addressing my other comments, please come back to this one and try reverting this change. I don't think it will be necessary anymore.

Reply:
Addressing the other commits the loop in not there anymore, anyway the issue still seems to persist.